src/Entity/DfProductBuild.php

```diff
@@ -128,46 +114,153 @@ private function buildProduct($product)
+        
+        foreach ($extraHeaders as $extraHeader) {
+            if (!empty($p[$extraHeader])) {
+                continue;
+            }
+            $p[$extraHeader] = isset($product[$extraHeader]) ? DfTools::cleanString($product[$extraHeader]) : '';
         }
 
         return $p;
     }
 
+    /**
+     * Applies specific transformations to a product's data for CSV export.
+     *
+     * This method performs several modifications on the product array, like:
+     * - If multi-price is enabled, formats the multiprice field.
+     * - Joins category values using a predefined separator.
+     * - If variants information exists, it slugifies and joins them using "%%".
+     * - Casts the group leader flag to an integer.
+     * - Iterates over extra headers to process attribute values:
+     *   - For each non-empty extra header, it concatenates key-value pairs in the format "key=value",
+     *     with the value cleaned and any "/" characters escaped.
+     * - Processes features (if present and an array) by converting them into an attributes string
+     *   formatted as "key=value" pairs joined with "/". The original features key is removed.
+     * - Ensures the final product fields are ordered according to the given headers.
+     *
+     * @param array $product      The associative array representing the product data.
+     * @param array $extraHeaders An array of additional headers to process in the product data.
+     * @param array $allHeaders   An array specifying the order of CSV fields.
+     *
+     * @return array The transformed product array ready for CSV export.
+     */
+    public function applySpecificTransformationsForCsv($product, $extraHeaders, $allHeaders)
+    {
+        if ($this->multipriceEnabled) {
+            $product['df_multiprice'] = DfTools::getFormattedMultiprice($product['df_multiprice']);
+        }
+        $product['categories'] = implode(DfTools::CATEGORY_SEPARATOR, $product['categories']);
+
+        if (array_key_exists('df_variants_information', $product)) {
+            $product['df_variants_information'] = implode('%%', array_map([__NAMESPACE__ . '\DfTools', 'slugify'], $product['df_variants_information']));
+        }
+
+        $product['df_group_leader'] = (int)$product['df_group_leader'];
+
+        if (array_key_exists('features', $product) && is_array($product['features'])) {
+
+            $formattedAttributes = array_map(
+                function ($key, $value) {
+                    if (is_array($value)) {
+                        $keyValueToReturn = [];
+                        foreach ($value as $singleValue) {
+                            $keyValueToReturn[] = $key . '=' . $singleValue;
+                        }
+                        return implode('/', $keyValueToReturn);
+                    }
+                    return $key . '=' . $value;
+                },
+                array_keys($product['features']),
+                $product['features']
+            );
```

Comment:
Buenas David! Quizá porque no estoy tan familiarizado con las funciones de PHP me había costado entender más esto. Dejo la versión con `foreach` por si crees que, ya que tratamos con key => value's, la hace más seguible de primeras. Si no, veo bien la versión que hay ahora :).

```suggestion
$formattedAttributes = [];

foreach ($product['features'] as $key => $value) {
    if (is_array($value)) {
        $keyValueToReturn = [];
        foreach ($value as $singleValue) {
            $keyValueToReturn[] = $key . '=' . $singleValue;
        }
        $formattedAttributes[] = implode('/', $keyValueToReturn);
    } else {
        $formattedAttributes[] = $key . '=' . $value;
    }
}
```

Reply:
Bueno, como ambas son válidas en verdad lo cambio :raised_hands:

Reply:
Cambiado en https://github.com/doofinder/doofinder-prestashop/pull/291/commits/b45697da9e08ccb89066336617e96ae12cee7beb :smile: